### ISSUE TITLE
cannot create image via cli in ARM mode

### ISSUE THREAD

**Issue by akirsman** [Azure/azure-xplat-cli#3093]
This seems to be a missing functionality - there's no command in ARM mode to upload nor register a .vhd as a VM image.

**Comment by amarzavery** [comment]
@huangpf @sauryadas - Can you take a look at this feature request and provide a timeline for the same?

**Comment by harshitha-bt** [comment]
Is there any update on this? If not through azure cli, is there any other way of achieving the same? I am using mac and hence powershell is not an option.

**Comment by huangpf** [comment]
There is a Capture API: https://msdn.microsoft.com/en-us/library/azure/mt163560.aspx, and there will be a new API for Creating VM images soon.

But note that to upload/download VHDs, you need to leverage the storage commands. Thanks.

**Comment by harshitha-bt** [comment]
Capture API is different from what is asked for, if i am right. Once the vhd is uploaded using storage commands, converting it to image is what is a missing functionality.

**Comment by huangpf** [comment]
@harshitha-bt You're correct. But CRP API should allow you to create a VM by directly attaching the VHD. Given that you can duplicate the VHD using storage commands, does that mitigate your problem? 

https://msdn.microsoft.com/en-us/library/azure/mt163591.aspx

```
...      "osDisk": {
        "name": "myosdisk1",
        "vhd": {
          "uri": "http://mystorage1.blob.core.windows.net/vhds/myosdisk1.vhd"
        },
        "caching": "ReadOnly | ReadWrite", // Changing this value causes VM to reboot
        "createOption": "fromImage | attach", ...
```

**Comment by hassanbabaie** [comment]
@huangpf can you shed any more light on what this means "API for Creating VM images soon" do we have an approx timeline and/or hint of what will change (become simpler?)

**Comment by hassanbabaie** [comment]
@huangpf can you let me know if this API was what you were talking about regarding coming soon?

2016-04-30-preview

https://github.com/Azure/azure-rest-api-specs/tree/master/arm-compute/2016-04-30-preview/swagger

Thanks

Hass

**Comment by huangpf** [comment]
I suppose so. It's the new set of APIs that I referred to, and I think it shall address the problem. But CLI is not updated with it yet.
